lib/model/model.go

```diff
@@ -2432,42 +2441,33 @@ func (m *model) GlobalDirectoryTree(folder, prefix string, levels int, dirsonly
 
 		f.Name = strings.Replace(f.Name, prefix, "", 1)
 
-		var dir, base string
-		if f.IsDirectory() && !f.IsSymlink() {
-			dir = f.Name
-		} else {
-			dir = filepath.Dir(f.Name)
-			base = filepath.Base(f.Name)
-		}
-
 		if levels > -1 && strings.Count(f.Name, sep) > levels {
 			return true
 		}
 
-		last := output
-		if dir != "." {
-			for _, path := range strings.Split(dir, sep) {
-				directory, ok := last[path]
-				if !ok {
-					newdir := make(map[string]interface{})
-					last[path] = newdir
-					last = newdir
-				} else {
-					last = directory.(map[string]interface{})
-				}
-			}
+		fullName := strings.Split(f.Name, sep)
+		name := fullName[len(fullName)-1]
+		path := fullName[:len(fullName)-1]
+
+		tree := DirectoryTree{Name: name, IsDirectory: f.IsDirectory(), Children: []*DirectoryTree{}}
+		for !reflect.DeepEqual(path, currentPrefix) && len(currentPrefix) != 0 {
```

Comment:
I'd be more comfortable doing string prefix comparison (we hust have to make sure not to match `foo/bar` against `foo/barbarian`)

Reply:
Do you have some reasonable argument why is matching prefixes better? I understand, why is string comparison better than comparing slices of strings. But why prefix matching?

In case of string matching, I had to care about delimiters which here and there caused a spaghetti code. I realized, that path won't be longer than some small constant, therefore I decided to use slice instead of strings. IMHO, the code is better without caring about delimiters... Is that really a problem?

Reply:
Essentially, the thing I don't like is reflect. I guess I meant equality comparison but said prefix comparison.
I am ok to compare string slices, but just not using reflect.

Reply:
Ok, I understand.
In my new solution, I still use a slice, but when comparing `currentPrefix` with the prefix of a given `FolderItem`, I create a string from `currentPrefix`. 

```
for strings.Join(currentPrefix, sep) != path && len(currentPrefix) != 0 {
	breadcrumbs = breadcrumbs[:len(breadcrumbs)-1]
	currentPrefix = currentPrefix[:len(currentPrefix)-1]
}
```
It is still in O(n^2) time (as before with reflect), we do not use reflect and does not have to care about delimiters.

Reply:
So you intend to compare it characterwise?
Yes, this looks more wasteless, however, I do not like the idea, where I should care about delimiters. IMHO, it is less readable. 
Anyway, isn't that a bit overoptimization? The file depth will really be small.